torch/csrc/jit/passes/alias_analysis.cpp

```diff
@@ -51,7 +51,8 @@ bool AliasDb::isContainerType(const TypePtr& type) {
 
 AliasDb::~AliasDb() = default;
 
-AliasDb::AliasDb(std::shared_ptr<Graph> graph) : graph_(std::move(graph)) {
+AliasDb::AliasDb(std::shared_ptr<Graph> graph, bool isFrozen)
+    : graph_(std::move(graph)), isFrozen_(isFrozen) {
```

Comment:
I don't think `prim::FrozenGetAttr` is right either.

Reply:
Oh yeah, we need to work out what happens when two attributes (`self.a` and `self.b`) alias. The current patch does not handle this case and potentially produces incorrect results by falsely inlining.

Reply:
Added a bunch of tests. Also Filed a bug because JIT is NOT preserving aliasing in __init (e.g. self.a = self.b)! 
Thanks Elias,  All your conditions are applied through preserveAttrs_  hashing.

Reply:
It's not preserving aliasing in lists, but it does with Tensors.

Reply:
Correct! this will be a follow up patch.